### ISSUE TITLE
NMS.GEN.LOC.004 - Method for end user to update location data

### ISSUE THREAD

**Issue by larubbio** [motech-implementations/mim#100]
The NMS system shall provide means to correct details of location data through a web interface via CSV upload or MDS UI.

It should be possible to do changes in spelling or other details of location

**Comment by llavoie** [comment]
This should hopefully just work through MDS UI, should be validated as part of Sprint 3 UAT.

**Comment by frankhuster** [comment]
Is the correct way for me to check this issue off:
- install motech + NMS
- use the MDS UI
- confirm I can change a district's name (for example)

?

**Comment by llavoie** [comment]
Didn't we decide that we'd use CSV upload for all updates, rather than MDS
UI?

On Thu, Jun 4, 2015 at 10:15 AM, Frank Huster <notifications@github.com>
wrote:

> Is the correct way for me to check this issue off:
>
>    - install motech + NMS
>    - use the MDS UI
>    - confirm I can change a district's name (for example)
>
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/issues/100#issuecomment-108979630>
> .
>



-- 
Lauren Lavoie
Software Development Manager, Mobile Health Innovations
Grameen Foundation
2101 4th Ave | Suite 1550 | Seattle, WA 98121
206-325-6690 ext 254
llavoie@grameenfoundation.org | Skype:laurenlavoie
Facebook <http://www.facebook.com/StopPovertyNow>│Twitter
<http://twitter.com/grameenfdn>│YouTube
<http://www.youtube.com/user/grameen?blend=6&ob=5> | www.motechsuite.org
*Connecting the World's Poor to Their Potential*

**Comment by frankhuster** [comment]
OK, so that issue is an actual dev work item that we can start on after your initial commit, right?

**Comment by llavoie** [comment]
It shouldn't be blocked on my work (I'm working on MCTS beneficiary upload). Kamil's changes for location import were merged a while back.

**Comment by frankhuster** [comment]
Doh, I mistakenly thought you were working on initial CSV stuff.
So I could start working on that using Kamil's stuff, right?

On Thu, Jun 4, 2015 at 10:40 AM, Lauren Lavoie <notifications@github.com>
wrote:

> It shouldn't be blocked on my work (I'm working on MCTS beneficiary
> upload). Kamil's changes for location import were merged a while back.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/issues/100#issuecomment-108987421>
> .
>



-- 
Frank Huster
Mobile Health Innovations
Grameen Foundation
+1.206.325.6690 x201

**Comment by llavoie** [comment]
Actually, @larubbio corrected me on this ticket -- he thinks we *can* use MDS UI for this one (so perhaps no work to do). And all location stuff should be on hold right now because he's working on removing language location code.

**Comment by larubbio** [comment]
We were given a file for this:

https://gist.github.com/llavoie/0263a5e6f290685fc460

**Comment by larubbio** [comment]
Looking over the file it isn't sufficient to implement this issue.  

1) it doesn't contain all elements of the hierarchy (i.e. no taluka)
2) it is unclear why some lines are repeated
3) it is unclear what to do when status is invalid and their are FLWs or beneficiaries in that location.
4) we need clear guidance on what "other details of location" is

**Comment by larubbio** [comment]
This was discussed on the monday night call, but I forget the resolution.  I'll see if John remembers.

**Comment by llavoie** [comment]
John is out, but Pankaj might recall?

On Wed, Jul 8, 2015 at 10:20 AM, Rob LaRubbio <notifications@github.com>
wrote:

> This was discussed on the monday night call, but I forget the resolution.
> I'll see if John remembers.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/issues/100#issuecomment-119669304>
> .
>



-- 
Lauren Lavoie
Software Development Manager, Mobile Health Innovations
Grameen Foundation
2101 4th Ave | Suite 1550 | Seattle, WA 98121
206-325-6690 ext 254
llavoie@grameenfoundation.org | Skype:laurenlavoie
Facebook <http://www.facebook.com/StopPovertyNow>│Twitter
<http://twitter.com/grameenfdn>│YouTube
<http://www.youtube.com/user/grameen?blend=6&ob=5> | www.motechsuite.org
*Connecting the World's Poor to Their Potential*

**Comment by larubbio** [comment]
I think the resolution was that the csv file with updates would match the original.  So I think changing the implementation from strictly creating records to creating or updating records will resolve this.  

We will not support deletion, or moving a child to a new parent (in that case a new child would be created)

**Comment by pankajnagpal30** [comment]
Yes. It will be done through CSV uploads and format will be same as data availability from MCTS.

 

Regards,

 

Pankaj Nagpal

+91 9810754321
